### PR TITLE
This contains the fixes from various forked repo, you can merge it.

### PR DESCRIPTION
**Pull request by indykish**
@timrogers  You merge and send an upstream request. 
Refer my README for more comments.

I added `payment_methods` api as well.

### CODE REVIEW

**Comment by timrogers** [comment]
@indykish Nice work. I'll go through this this week - I'm keen to fork this gem as it's annoying not to be able to get things fixed.

**Comment by timrogers** [comment]
To be honest though, I'm quite keen to start again on this gem if I get some time and make it a bit nicer...

File: README.markdown [hunk 1]
```diff
@@ -15,13 +46,33 @@ whmcs-ruby provides Ruby bindings for the [WHMCS API](http://wiki.whmcs.com/API:
...
+Recently, WHMCS offered [another type](http://docs.whmcs.com/API:Access_Keys) of authenticating API requests.
+Using API Access Key might be useful for mobile devices and/or IPv6 bug WHMCS whitelists currently have.
+
+	require 'whmcs'
+
+    WHMCS.configure do |config|
+      config.api_url      = 'http://example.com/includes/api.php'
+      config.api_username = 'someusername'
+      config.api_password = 'c4ca4238a0b923820dcc509a6f75849b' # md5 hash
+	  config.api_access_key = 'YetAnotherAPIAccessKeyForWHMCS'
```
Comment (timrogers) [inline README.markdown]: Can you fix the indentation on this?

File: README.markdown [hunk 2]
```diff
@@ -15,13 +46,33 @@ whmcs-ruby provides Ruby bindings for the [WHMCS API](http://wiki.whmcs.com/API:
...
 details.
 
 
 ## Installation
 
-    gem install whmcs-ruby
+Caveat: `gem install whmcs-ruby` will not work as this isn't pushed yet.
+
+
+    gem 'whmcs-ruby', :git => 'https://github.com/megamsys/whmcs-ruby.git'
```
Comment (timrogers) [inline README.markdown]: Change this to `:github => "timrogers/whmcs-ruby"`

File: Rakefile [hunk 3]
```diff
@@ -14,18 +14,18 @@ task :console do
...
-
-  sh "gem build whmcs-ruby.gemspec"
-  sh "gem push whmcs-ruby-#{ver}.gem"
-  sh "git tag -a -m 'whmcs-ruby v#{ver}' v#{ver}"
-  sh "git push origin v#{ver}"
-  sh "git push origin master"
-  sh "git clean -fd"
-  sh "rake pages"
-end
+# require 'sdoc_helpers'
```
Comment (timrogers) [inline Rakefile]: Just delete this I think, rather than commenting it out

File: lib/whmcs/base.rb [hunk 4]
```diff
@@ -22,6 +23,11 @@ def self.send_request(params = {})
         :password => WHMCS.config.api_password
       )
 
+	  # alternative API access
+	  if( !WHMCS.config.api_access_key.nil? )
```
Comment (timrogers) [inline lib/whmcs/base.rb]: `unless` is nicer than `if !` - plus no need for brackets in Ruby :smile:

File: lib/whmcs/base.rb [hunk 5]
```diff
@@ -34,18 +40,20 @@ def self.send_request(params = {})
       req.set_form_data(params)
 
       res = http.start { |http| http.request(req) }
-      parse_response(res.body)
+      parse_response(HTMLEntities.new.decode(res.body))
```
Comment (timrogers) [inline lib/whmcs/base.rb]: I think the `HTMLEntities` call should probably occur in `#parse_response` - feels like something that is part of the parsing

File: lib/whmcs/client.rb [hunk 6]
```diff
@@ -322,6 +340,7 @@ def self.send_email(params = {})
       params.merge!(:action => 'sendemail')
       send_request(params)
     end
+<<<<<<< HEAD
```
Comment (timrogers) [inline lib/whmcs/client.rb]: Looks like a leftover from some conflicts

File: lib/whmcs/config.rb [hunk 7]
```diff
@@ -10,11 +10,15 @@ class Config
     # The WHMCS API URL
     attr_accessor :api_url
 
+	# The WHMCS API Access Key
```
Comment (timrogers) [inline lib/whmcs/config.rb]: Some odd indentation in this file too

**Comment by timrogers** [comment]
@indykish Good job! Made a bunch of small comments, but should be able to get this merged soon :heart:

File: whmcs-ruby.gemspec [hunk 8]
```diff
@@ -16,7 +16,8 @@ Gem::Specification.new do |s|
   s.files           += Dir['lib/**/*']
   s.files           += Dir['test/**/*']
 
-  s.add_dependency('crack', '= 0.1.8')
+  s.add_dependency('crack', '= 0.4.3')
+  s.add_dependency('htmlentities', '= 4.3.4')
```
Comment (timrogers) [inline whmcs-ruby.gemspec]: It'd be nicer not to bind to a particular version as it means that the gem can't be used in a project which uses any other version of these gems. Can we be a bit more permissive on this dependency?

File: README.markdown (hunk 1 above)
Comment (indykish) [inline README.markdown]: Done.  pull req. on its way.

File: Rakefile (hunk 3 above)
Comment (indykish) [inline Rakefile]: Done.  pull req. on its way.

File: README.markdown (hunk 2 above)
Comment (indykish) [inline README.markdown]: Done.  pull req. on its way.

File: lib/whmcs/base.rb (hunk 4 above)
Comment (indykish) [inline lib/whmcs/base.rb]: Done.  pull req. on its way.

File: lib/whmcs/base.rb (hunk 5 above)
Comment (indykish) [inline lib/whmcs/base.rb]: Done.

File: lib/whmcs/client.rb (hunk 6 above)
Comment (indykish) [inline lib/whmcs/client.rb]: Thanks, done. 

I am doing tests today for our project. https://github.com/megamsys/nilavu

File: lib/whmcs/config.rb (hunk 7 above)
Comment (indykish) [inline lib/whmcs/config.rb]: Done. Did `CTRL B' in`atom - ruby-beautify` :)

File: whmcs-ruby.gemspec (hunk 8 above)
Comment (indykish) [inline whmcs-ruby.gemspec]: Done,  changed it `~> xyz version`

**Comment by indykish** [comment]
@timrogers  I just grouped all the fixes. Credit goes to you/others in starting it up. :+1: 

I have fixed and taken cared  of your comments. Let me test it in our env. today.

**Pull request closed without merging**